chat-widget/src/components/webchatcontainerstateful/webchatcontroller/markdownrenderers/HyperlinkTextOverrideRenderer.ts

```diff
@@ -0,0 +1,63 @@
+import DOMPurify from "dompurify";
+
+class HyperlinkTextOverrideRenderer {
+    private hyperlinkTextOverride: boolean
+    constructor(hyperlinkTextOverride: boolean) {
+        this.hyperlinkTextOverride = hyperlinkTextOverride;
+    }
+
+    private convertTextToHtmlNode(text: string) {
+        const htmlNode = document.createElement("div");
+
+        try {
+            text = DOMPurify.sanitize(text); // eslint-disable-line @typescript-eslint/no-explicit-any
+            htmlNode.innerHTML = text;
+        } catch {
+            return htmlNode;
+        }
+
+        return htmlNode;
+    }
+
+    private processANode(htmlNode: any) { // eslint-disable-line @typescript-eslint/no-explicit-any
+        const aTags = htmlNode.getElementsByTagName("a");
+
+        for (let index = 0; index < aTags.length; index++) {
+            const aNode = aTags[index];
+
+            if (!aNode ||
+                !aNode.tagName || aNode.tagName.toLowerCase() !== "a" ||
+                !aNode.href) continue;
+            if (aNode.href !== aNode.innerText.trim()) {

```

Comment:
Do we need to compare this with casing ? 
WWW.BING.COM == www.bing.com ???

![image](https://user-images.githubusercontent.com/49963796/230974297-23ae19fb-545d-4075-81c6-ac4294588d35.png)

Reply:
`"WWW.BING.COM" == "www.bing.com"` should return `false`